app/api/resources/user.py

```diff
@@ -498,3 +498,99 @@ def get(cls):
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data["email"]:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid!={}:
+            return is_valid, HTTPStatus.BAD_REQUEST
+        
+        user = DAO.get_user_by_email(data["email"])
+ 
+        if user is None:
+            return messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM, HTTPStatus.NOT_FOUND
+ 
+        if not user.is_email_verified:
+            return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, HTTPStatus.FORBIDDEN    
+ 
+        send_email_reset_password_message(user.name, data["email"])
+ 
+        return messages.PASSWORD_RESET_MAIL_MESSAGE, HTTPStatus.OK
+ 
+@users_ns.route("user/reset_password")
+@users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_SUCCESSFULLY_UPDATED}")
+@users_ns.response(
+    HTTPStatus.UNAUTHORIZED,
+    f"{messages.RESET_PASSWORD_TOKEN_HAS_EXPIRED}"'\n'f"{messages.AUTHORISATION_TOKEN_IS_MISSING}"
+)
+@users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.PASSWORD_FIELD_IS_MISSING}")    
+ 
+class ResetPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"reset")
+    @users_ns.expect(reset_password_forgot_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password
+ 
+        This will allow the user to change/create a new password 
+        In case they forgot it.
+ 
+        Args:
+        token: token of reset link sent on verified mail
+        new_password: take the new validated password to login
+ 
+        Return:
+        Password Updated successfully
+        """
+        data = request.json
+       
+        if not data["token"]:
+            return messages.AUTHORISATION_TOKEN_IS_MISSING, HTTPStatus.UNAUTHORIZED
+ 
+        if not data["new_password"] :
+            return messages.PASSWORD_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST    
+ 
+        is_valid = validate_reset_password(data)
+ 
+        if is_valid != {}:
+            return is_valid, HTTPStatus.BAD_REQUEST
+ 
+        email_from_token = DAO.reset_password_token(data["token"])
+ 
+        if not email_from_token :
+            return messages.RESET_PASSWORD_TOKEN_HAS_EXPIRED, HTTPStatus.UNAUTHORIZED
+ 
+        user = DAO.get_user_by_email(email_from_token)
```

Comment:
could you include these 4 lines in a DAO function? you are using the capabilities of `UserModel` in the API level (when you do `user.save_to_db()`). The idea of DAO is that it is the bridge between `UserModel` and `UserAPI`

Reply:
Ok

Reply:
Hi @isabelcosta sorry for not being active for sometime, was really busy but now  i am adding all the suggested changes in the commit but can you please explain what all four lines i need to add in DAO ? did you mean by this "if condition"? sorry i tried but i can't find way to do so ? can you please help if you are not busy please!! ?

Reply:
or should i create a new function in DAO of user ? @isabelcosta . as right now , i can't see anyother way to add these lines [user.save_to_db] to DAO , although currently there is a function in DAO already named -- "change_password" but it takes two passwords (current & new) as it's arguments whereas in my case i am just taking the new password as input from user to change as the user doesn't know current password and thats why applying for forget password ??? can you please reply quickly if you've time so, that i can add the changes soon... as now its too long to my this PR .. ?